### PR TITLE
Release Notes 10/25

### PR DESCRIPTION
**Pull request by alvinli222**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,34 @@
...
+  * AKS version [`1.31`](https://kubernetes.io/blog/2024/08/13/kubernetes-v1-31-release/) is now generally available. Please check the release tracker for when your region will receive the GA update. Some regions may not receive this update until later in November.
+  * The first official patch version of LTS 1.27 is being released.
+ 
+* Preview features:
+  * We've added a new way to optimize your upgrade process drain behavior. By default, a node drain failure causes the upgrade operation to fail, leaving the undrained nodes in a schedulable state, this behavior is called `Schedule`. Alternatively, you can select the `Cordon` behavior, which skips nodes that fail to drain by placing them in a quarantined state, labeling them `kubernetes.azure.com/upgrade-status:Quarantined` and proceeds with upgrading the remaining nodes. This ensures that all nodes are either upgraded or quarantined. This approach allows you to troubleshoot drain failures and gracefully manage the quarantined nodes.
+    
+* Bug fixes:
+ 
+* Behavior change:
+
```
Comment (sdesai345) [inline CHANGELOG.md:28]: * You can now fine-tune supported models on KAITO version [`0.3.1`](https://github.com/Azure/kaito/releases/tag/v0.3.1) with the AI toolchain operator add-on on your AKS cluster.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,60 @@
 # Azure Kubernetes Service Changelog
 
+## Release 2024-10-25
+
+Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/). This release is titled as `v20241025`.
+
+### Announcements
+
+* AKS version 1.30 is now available as a [Long term support version](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar) and AKS version 1.28 End of Life is Jan, 15 2025.
+* Upgrade from LTS 1.27 to LTS 1.30 is now supported.
```
Comment (shashankbarsin) [inline CHANGELOG.md:10]: These were present in the previous release note too. Maybe we can remove the second one. For first one, should we just keep the 1.28 EOL part?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,60 @@
...
+* AKS will start to block new cluster creation with basic load balancer in January 2025. Basic Load Balancer will be deprecated September 31 2025 and affected clusters must be migrated to the Standard Load Balancer prior to that date. Refer to [BLB deprecation announcement](
+https://azure.microsoft.com/updates/azure-basic-load-balancer-will-be-retired-on-30-september-2025-upgrade-to-standard-load-balancer/) for more information.
+
+### Release Notes
+
+* Features:
+  * AKS patch versions 1.28.14, 1.29.9, 1.30.5 are now available. Refer to [version support policy](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#kubernetes-version-support-policy) and [upgrading a cluster](https://learn.microsoft.com/azure/aks/upgrade-aks-cluster?tabs=azure-cli) for more information.
+  * AKS version [`1.31`](https://kubernetes.io/blog/2024/08/13/kubernetes-v1-31-release/) is now generally available. Please check the release tracker for when your region will receive the GA update. Some regions may not receive this update until later in November.
+  * The first official patch version of LTS 1.27 is being released.
```
Comment (shashankbarsin) [inline CHANGELOG.md:22]: can we include the exact patch version string itself?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,60 @@
...
+ 
+* Preview features:
+  * We've added a new way to optimize your upgrade process drain behavior. By default, a node drain failure causes the upgrade operation to fail, leaving the undrained nodes in a schedulable state, this behavior is called `Schedule`. Alternatively, you can select the `Cordon` behavior, which skips nodes that fail to drain by placing them in a quarantined state, labeling them `kubernetes.azure.com/upgrade-status:Quarantined` and proceeds with upgrading the remaining nodes. This ensures that all nodes are either upgraded or quarantined. This approach allows you to troubleshoot drain failures and gracefully manage the quarantined nodes.
```
Comment (shashankbarsin) [inline CHANGELOG.md:25]: docs link?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,60 @@
...
+  * You can now [block pod access to the Azure Instance Metadata Service (IMDS) endpoint](https://learn.microsoft.com/azure/aks/imds-restriction) to enhance security.
+    
+* Bug fixes:
```
Comment (shashankbarsin) [inline CHANGELOG.md:28]: this section is empty. we can remove this section if we don't have any points to include under here

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,60 @@
...
+ 
+* Behavior change:
+  * You can now fine-tune supported models on KAITO version [0.3.1](https://github.com/Azure/kaito/releases/tag/v0.3.1) with the AI toolchain operator add-on on your AKS cluster.
+  * Beta APIs will be disabled for newly created clusters after November 30, 2024.
```
Comment (shashankbarsin) [inline CHANGELOG.md:32]: from a specific K8s version or API version?

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,60 @@
...
+  * Virtual Machine nodepools will not be able to use custom VNets when if the cluster is using system-assigned identity.
+  * LTS no longer do defaulting on autoUpgrader profile.
+  * This release adds the below plugin blocks to coredns corefile by default.
+    * `template ANY ANY reddog.microsoft.com {      rcode NXDOMAIN    }`
+    * `template ANY ANY internal.cloudapp.net {      match "^(?:+.){4,}internal.cloudapp.net.$"      rcode NXDOMAIN      fallthrough    }`
+    * The first plugin block is designed to return NXDOMAIN for queries to reddog.microsoft.com domain. This is useful to prevent coredns from sending queries to AzureDNS for reddog.microsoft.com domain.
+    * The second plugin block is designed to return NXDOMAIN for queries to internal.cloudapp.net domain, except for domains with the required number of components following this pattern: ..*.internal.cloudapp.net. Example - Allowed: aks-agentpool-xxxx-vmssxxxxxx.xxxxxx.xx.internal.cloudapp.net . Example - Not allowed: test.aks-agentpool-xxxx-vmssxxxxxx.xxxxxx.xx.internal.cloudapp.net
+  
+* Component updates:
+  * The KEDA addon will be bumped to version 1.15 on AKS 1.32 instead of AKS 1.31.
```
Comment (shashankbarsin) [inline CHANGELOG.md:42]: this should be under announcements. in this section we can only call out components which got updated as part of this AKS release

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,60 @@
...
+  * The CoreDNS addon will use v1.9.4-2 for all existing clusters and clusters created for versions up until 1.30. For kubernetes version 1.31+, CoreDNS will use v1.11.3-2
```
Comment (shashankbarsin) [inline CHANGELOG.md:43]: nit: can we change 'will use' to 'updated to' for consistency with format of component updates points from previous releases?

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,60 @@
...
+  * The Image Cleaner addon will now use v1.4.0
```
Comment (shashankbarsin) [inline CHANGELOG.md:44]: nit: same comment as above. can we change 'will use' to 'updated to'?

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,60 @@
...
+* Starting on January 1, 2025, [invalid values sent to the Azure AKS API for the properties.mode field of AKS AgentPools will be rejected](https://github.com/Azure/AKS/issues/4468). Prior to this change, unknown modes were assumed to be User. The only valid values for this field are the (case-sensitive) strings:["User", "System"](https://learn.microsoft.com/rest/api/aks/agent-pools/create-or-update?view=rest-aks-2024-02-01&tabs=HTTP#agentpoolmode), or ["Gateway"](https://learn.microsoft.com/rest/api/aks/agent-pools/create-or-update?view=rest-aks-2024-06-02-preview&tabs=HTTP#agentpoolmode).
+* AKS will start to block new cluster creation with basic load balancer in January 2025. Basic Load Balancer will be deprecated September 31 2025 and affected clusters must be migrated to the Standard Load Balancer prior to that date. Refer to [BLB deprecation announcement](
+https://azure.microsoft.com/updates/azure-basic-load-balancer-will-be-retired-on-30-september-2025-upgrade-to-standard-load-balancer/) for more information.
+
+### Release Notes
+
+* Features:
```
Comment (qpetraroia) [inline CHANGELOG.md:19]: Can we add

[GitHub Copilot](https://azure.microsoft.com/en-us/products/github/Copilot?msockid=39613fded86166a01c612b53d9db67d8) for Azure now supports [AKS commands](https://learn.microsoft.com/en-us/azure/aks/aks-extension-ghcopilot-plugins).

File: CHANGELOG.md (hunk 2 above)
Comment (kaarthis) [reply CHANGELOG.md:10]: Lets move 1.28 EOL to its own line item and please put EOL date as Jan 31 2025 (likely to be end of Jan).

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,60 @@
...
+* Starting on January 1, 2025, [invalid values sent to the Azure AKS API for the properties.mode field of AKS AgentPools will be rejected](https://github.com/Azure/AKS/issues/4468). Prior to this change, unknown modes were assumed to be User. The only valid values for this field are the (case-sensitive) strings:["User", "System"](https://learn.microsoft.com/rest/api/aks/agent-pools/create-or-update?view=rest-aks-2024-02-01&tabs=HTTP#agentpoolmode), or ["Gateway"](https://learn.microsoft.com/rest/api/aks/agent-pools/create-or-update?view=rest-aks-2024-06-02-preview&tabs=HTTP#agentpoolmode).
+* AKS will start to block new cluster creation with basic load balancer in January 2025. Basic Load Balancer will be deprecated September 31 2025 and affected clusters must be migrated to the Standard Load Balancer prior to that date. Refer to [BLB deprecation announcement](
```
Comment (kaarthis) [inline CHANGELOG.md:14]: Should we put Jan 31, 2025 to be precise here.

File: CHANGELOG.md (hunk 3 above)
Comment (kaarthis) [reply CHANGELOG.md:22]: Yes i believe its 1.27.100 .

File: CHANGELOG.md (hunk 4 above)
Comment (kaarthis) [reply CHANGELOG.md:25]: https://github.com/MicrosoftDocs/azure-aks-docs-pr/pull/343/files waiting on resolving a build issue .

File: CHANGELOG.md (hunk 6 above)
Comment (kaarthis) [reply CHANGELOG.md:32]: Yes good point the PR says for cluster creations in version 1.28, 1.29 might have to double check the PR and add it in @Alvinli

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,60 @@
...
+    * `template ANY ANY internal.cloudapp.net {      match "^(?:+.){4,}internal.cloudapp.net.$"      rcode NXDOMAIN      fallthrough    }`
+    * The first plugin block is designed to return NXDOMAIN for queries to reddog.microsoft.com domain. This is useful to prevent coredns from sending queries to AzureDNS for reddog.microsoft.com domain.
+    * The second plugin block is designed to return NXDOMAIN for queries to internal.cloudapp.net domain, except for domains with the required number of components following this pattern: ..*.internal.cloudapp.net. Example - Allowed: aks-agentpool-xxxx-vmssxxxxxx.xxxxxx.xx.internal.cloudapp.net . Example - Not allowed: test.aks-agentpool-xxxx-vmssxxxxxx.xxxxxx.xx.internal.cloudapp.net
+  
+* Component updates:
+  * The KEDA addon will be bumped to version 1.15 on AKS 1.32 instead of AKS 1.31.
+  * The CoreDNS addon will use v1.9.4-2 for all existing clusters and clusters created for versions up until 1.30. For kubernetes version 1.31+, CoreDNS will use v1.11.3-2
+  * The Image Cleaner addon will now use v1.4.0
+
+ TO UPDATE:
```
Comment (kaarthis) [inline CHANGELOG.md:46]: Why do we say To update . Its actually happening right so it should be 'Component updates' as per other rel notes.

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,60 @@
...
+  * upgrade blob csi driver versions upgrade to v1.22.9 on aks 1.27
```
Comment (kaarthis) [inline CHANGELOG.md:47]: Instead of Upgrade lets say 'Upgraded'.

**Review by kaarthis — COMMENTED** [review]
PTAL

File: CHANGELOG.md (hunk 10 above)
Comment (haitch) [reply CHANGELOG.md:19]: yep, let's add a entry for AKS runCommand:
- we bump couple binary version: kubectl to 1.30.5, helm to 3.15.4
- also added couple new binary: jq, awk, grep, xargs, fix https://github.com/Azure/AKS/issues/3034 
- new copilot support

**Comment by phealy** [comment]
Closing in favor of https://github.com/Azure/AKS/pull/4621/files

**Pull request closed without merging**